src/main/java/bio/terra/testrunner/runner/TestRunner.java

```diff
@@ -604,24 +641,28 @@ public static boolean runTest(String configFileName, String outputParentDirName)
 
       // get an instance of a runner and tell it to execute the configuration
       TestRunner runner = new TestRunner(testConfiguration);
+      boolean testConfigFailed = false;
       try {
         runner.executeTestConfiguration();
 
-        // update the failure flag if it's not already been set
-        if (!isFailure) {
-          for (TestScriptResult.TestScriptResultSummary testScriptResultSummary :
-              runner.summary.testScriptResultSummaries) {
-            if (testScriptResultSummary.isFailure) {
-              isFailure = true;
-              break;
-            }
+        // even if the test configuration didn't throw an exception, it still may have failed due to
+        // a timeout
+        for (TestScriptResult.TestScriptResultSummary testScriptResultSummary :
+            runner.summary.testScriptResultSummaries) {
+          if (testScriptResultSummary.isFailure) {
+            testConfigFailed = true;
+            break;
           }
         }
       } catch (Exception runnerEx) {
         logger.error("Test Runner threw an exception", runnerEx);
-        isFailure = true;
+        testConfigFailed = true;
       }
 
+      // update the failure flag for this test config and the whole suite
+      testConfigNameToResult.put(testConfiguration.name, testConfigFailed);
```

Comment:
maybe rename this `testConfigNameToFailed` for clarity, or use an enum of PASSED/FAILED in the map instead of a boolean.

Reply:
Changed the variable name.